python/tvm/relay/frontend/pytorch.py

```diff
@@ -110,12 +102,20 @@ def inplace_add_to_add(op_name):
     if len(intersect) > 0 and intersect != set(["aten::add"]):
         return True
 
-    if is_used_by_list_add(filter(lambda use: use.user.kind() != "prim::Loop", uses)):
-        return True
+    # if add op outputs list, it is dynamic so we need to construct List ADT
+    for use in filter(lambda use: use.user.kind() in ["aten::add", "aten::add_"], uses):
```

Comment:
I'm a bit weary that this is very brittle. Probably not more than before, though.

Reply:
Yes, with this PR `_should_construct_dynamic_list` runs more often. I got an error from this function when `_get_node_type` was called on a op which returns multiple output. I've cleaned up this function a bit so that `_get_node_type` will only run on `aten::add` and `aten::add_`. So in that sense it is no worse than before.

I agree that this function is a bit hacky. I added this when I was trying to support Python list. It works on the test cases I was working on, but probably it is not robust.